### DISCUSSION TITLE
Treatment Missing Value

### DISCUSSION THREAD

**Discussion by ToniAS21** [teamalgoritma/community#171] [Q&A] [answered]
Permisi kakak2 semua, saya mau nanya bagaimana sebaiknya treatment yang diambil pada data yang memiliki banyak "Unknown" ? Dataset ini mau digunakan dalam membuat clustering. 

- Berhubung data kosongnya berupa kategorikal apakah sebaiknya menggunakan modus data saja? Tapi saya rasa akan memberikan pengaruh signifikan buruk terhadap hasil clustering karena datanya banyak bukan asli dari narasumber.

- Atau sebaiknya dihapus saja baris atau kolom tersebut? tetapi saya rasa dapat menimbulkan berkurangnya data dalam jumlah banyak.

![Screenshot (1582)](https://user-images.githubusercontent.com/83503249/176993725-1c6c45b8-5e29-4268-81eb-04d4879751f1.png)

Sekian terima kasih

**Comment by triarm** [comment, marked as answer]
Hi @ToniAS21 ,

Pada kasus data seperti itu, berikut beberapa treatment yang dapat dilakukan:
1. Menghapus baris/observasi yang mengandung "Unknown". 
2. Jika banyak data yang bersih menjadi jauh berkurang, maka Anda dapat menghapus kolom-kolom yang mengandung unknown dengan asumsi kolom tersebut tidak memiliki pengaruh yang signifikan terhadap analisis.
3. Jika kolom yang akan dihapus ternyata memiliki pengaruh yang signifikan maka metode lainnya yang dapat dilakukan adalah mengisi nilai Unknown tersebut dengan modus dari kolomnya atau mendefinisikan Unknown tersebut adalah kategori baru (Jika dilihat dari data tersebut, sepertinya kolom yang mengandung Unknown adalah kolom yang memiliki tipe kategori)

Anda dapat mencoba beberapa metode di atas dan memilih metode mana yang sesuai dengan kebutuhan analisis Anda atau metode yang menghasilkan cluster terbaik.
